src/components/NotFound.astro

```diff
@@ -2,41 +2,24 @@
 import { Image } from 'astro:assets';
 import Houston from '~/assets/houston_astronaut.png';
 import NotFoundSvg from '~/assets/not_found.png';
-import languages from '~/i18n/languages';
 import type { UILanguageKeys } from '~/i18n/translation-checkers';
 import { useTranslationsForLang } from '~/i18n/util';
 
-const defaultLang = 'en';
-const t = useTranslationsForLang(defaultLang);
-
-/**
- * Dictionary of translated content for the 404 page.
- * Creates an object like:
- * ```js
- * {
- *   en: { p: 'This page isn’t in our solar system.', a: 'Take me home.' },
- *   // other languages
- * }
- * ```
- * `p` targets the paragraph content, `a` targets the link text.
- */
-const dictionary = Object.fromEntries(
-	Object.keys(languages).map((_lang) => {
-		const lang = _lang as UILanguageKeys;
-		const t = useTranslationsForLang(lang);
-		return [lang, { p: t('404.content'), a: t('404.linkText') }];
-	})
-);
+interface Props {
+	lang: UILanguageKeys;
+}
+const { lang } = Astro.props;
+const t = useTranslationsForLang(lang);
 ---
 
 <div class="wrapper">
 	<Image class="galaxy" src={NotFoundSvg} alt="" loading="eager" />
 	<Image class="houston" src={Houston} alt="" loading="eager" width="192" />
-	<four-oh-four data-dictionary={JSON.stringify(dictionary)}>
+	<four-oh-four>
 		<div class="hero-text">
 			<h1>404</h1>
 			<p>{t('404.content')}</p>
-			<a href={`/${defaultLang}/`}>{t('404.linkText')}</a>
+			<a href={`/${lang}/`}>{t('404.linkText')}</a>
```

Comment:
The link on the 404 page goes to the 404 page (You can even tell by hovering that it's pointing to `/404`, it's not a redirect)

![image](https://github.com/withastro/docs/assets/5098874/479a81c6-2280-4ee0-ac06-d555d7b91f0f)